techmatters/flex-plugins
# Audio Notifications for Reserved tasks and new messages

Primary reviewer: @stephenhand 

## Description
- This PR adds audio notifications - a repeating alert when a task gets reserved, and a single beep(bell) when the counsellor receives a  new message. 
- Introduces `notifyReservedTask` that will trigger an alert based on workerClient's `reservationCreated` listener and `notifyNewMessage` that will trigger an alert based on conversationsClient's `messageAdded` listener.
- In a separate PR, Bo [added](https://github.com/techmatters/infrastructure-config/pull/112) an s3 bucket for each environement with Cloudfront for static assets. 
- To use this, new service configuration attribute has been added called `assets_bucket_url`. For all helplines, configurations will need to be updated with `assets_bucket_url` to ensure audio notifications work. 

### Checklist
- [n/a] New tests added
- [n/a] Feature flags added
- [n/a] Strings are localized
- [x] Added a new property in Service Configuration

### Related Issues
Fixes #[1619](https://tech-matters.atlassian.net/browse/CHI-1619)

### Verification steps
- Start a conversation in webchat or other chat channels. 
- Ensure Flex is active in browser and audio is turned on. 
- For checking reserved task notification, change to 'Available' as worker and ensure no other tasks are active. A continuous beep should persist. To stop the alert, change status of the worker or task
- For checking new message notification, (this is tricky - use a phone or other device to ensure flex is active) send new messages from the chat channel. A single bell sound should happen.
